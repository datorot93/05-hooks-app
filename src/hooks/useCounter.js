import { useState } from 'react';


export const useCounter = ( initialState = 10) => {
    const [state, setState] = useState(initialState);

    const increment = ( steps = 1) => {
        setState( state + steps );
    };

    const decrement = ( steps = 1) => {
        setState( state - steps );
    };

    const reset = () => {
        setState( initialState )
    }

    return {
        state,
        increment,
        decrement,
        reset
    }
}
