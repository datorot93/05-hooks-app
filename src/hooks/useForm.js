import { useState } from "react"

// El initialState es el objeto que se declaraba en el useState del useState de SimpleForm.js
export const useForm = ( initialState = {} ) => {

    const [values, setValues] = useState(initialState);

    const reset = () => {
        setValues( initialState );
    }

    const handleInputChange = ({ target }) => {

        setValues({
            ...values,
            [ target.name ]: target.value
        });
    }

    return [ values, handleInputChange, reset ];
}

