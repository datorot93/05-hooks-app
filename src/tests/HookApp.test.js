import React from 'react';
import { shallow } from 'enzyme';
import { HookApp } from '../hooks/HookApp';


describe('Prueba en HookApp', () => {
    test('debe tomar una snapshoot de mi HookApp', () => {
        const wrapper = shallow( <HookApp /> );
        expect( wrapper ).toMatchSnapshot();

    })
    
})
