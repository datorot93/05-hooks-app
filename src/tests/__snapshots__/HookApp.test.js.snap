// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en HookApp debe tomar una snapshoot de mi HookApp 1`] = `
<div>
  <h1>
    Hola Mundo
  </h1>
</div>
`;
